api/commands/run.ts

```diff
@@ -118,7 +120,19 @@ export class Runner {
 
       // Currently, we don't support caching for operations (and any dependents)
       await this.adapter.persistStateMetadata(
-        successfulActions.filter(action => action.type !== "operation")
+        successfulActions.filter(action => action && action.type !== "operation")
+      );
+      await this.adapter.deleteStateMetadata(
```

Comment:
what's the risk of this failing? if the metadata fails to be deleted, will the next run be able to run correctly?